src/components/ScreenWrapper/BaseScreenWrapper.js

```diff
@@ -0,0 +1,159 @@
+            this.unsubscribeTransitionEnd();
+        }
+        if (this.beforeRemoveSubscription) {
+            this.beforeRemoveSubscription();
+        }
+    }
+
+    render() {
+        return (
+            <SafeAreaConsumer>
+                {({
+                    insets, paddingTop, paddingBottom, safeAreaPaddingBottomStyle,
+                }) => {
+                    const paddingStyle = {};
+
+                    if (this.props.includePaddingTop) {
+                        paddingStyle.paddingTop = paddingTop;
+                    }
+
+                    // We always need the safe area padding bottom if we're showing the offline indicator since it is bottom-docked.
+                    if (this.props.includeSafeAreaPaddingBottom || this.props.network.isOffline) {
+                        paddingStyle.paddingBottom = paddingBottom;
+                    }
+
+                    return (
+                        <View
+                            style={[
+                                ...this.props.style,
+                                styles.flex1,
+                                paddingStyle,
+                            ]}
+                            // eslint-disable-next-line react/jsx-props-no-spreading
+                            {...(this.props.environment === CONST.ENVIRONMENT.DEV) ? this.panResponder.panHandlers : {}}
+                        >
+                            <KeyboardAvoidingView style={[styles.w100, styles.h100, {maxHeight: this.props.windowHeight}]} behavior={this.props.keyboardAvoidingViewBehavior}>
+                                <HeaderGap />
+                                {(this.props.environment === CONST.ENVIRONMENT.DEV) && <TestToolsModal />}
+                                {// If props.children is a function, call it to provide the insets to the children.
+                                    _.isFunction(this.props.children)
+                                        ? this.props.children({
+                                            insets,
+                                            safeAreaPaddingBottomStyle,
+                                            didScreenTransitionEnd: this.state.didScreenTransitionEnd,
+                                        })
+                                        : this.props.children
+                                }
+                                {this.props.isSmallScreenWidth && (
+                                    <OfflineIndicator />
+                                )}
+                            </KeyboardAvoidingView>
+                        </View>
+                    );
+                }}
+            </SafeAreaConsumer>
+        );
+    }
+}
+
+BaseScreenWrapper.propTypes = propTypes;
+BaseScreenWrapper.defaultProps = defaultProps;
+
+export default compose(
+    withNavigation,
+    withKeyboardState,
+    withOnyx({
+        modal: {
+            key: ONYXKEYS.MODAL,
+        },
+        isTestToolsModalOpen: {
```

Comment:
I can't find the usage of `isTestToolsModalOpen` prop.

Reply:
It is used to get value from Onyx, see action: src/libs/actions/TestTool.js

Reply:
+1 please remove that unused prop

Reply:
@alexxxwork I still didn't understand where is it being used in `ScreenWrapper` component. I think we don't need to read the value from Onyx here. We're already doing that in `actions/TestTool` right?

Reply:
Yeah it's not needed here